carma_wm/src/collision_detection.cpp

```diff
@@ -0,0 +1,179 @@
+#include "carma_wm/collision_detection.h"
+
+namespace carma_wm {
+
+    namespace collision_detection {
+
+        std::vector<cav_msgs::RoadwayObstacle> CollisionChecking::WorldCollisionDetection(cav_msgs::RoadwayObstacleList& rwol, cav_msgs::TrajectoryPlan& tp, geometry_msgs::Vector3& size, geometry_msgs::Twist& veloctiy, int target_time){
+
+            std::vector<cav_msgs::RoadwayObstacle> rwo_collison;
+
+            collision_detection::MovingObject vehicle_object = CollisionChecking::ConvertVehicleToMovingObject(tp, size, veloctiy);
+
+            for (int i = 0; i < rwol.roadway_obstacles.size(); i++) {
+
+                collision_detection::MovingObject rwo = CollisionChecking::ConvertRoadwayObstacleToMovingObject(rwol.roadway_obstacles[i]);
+
+                bool collision = CollisionChecking::DetectCollision(vehicle_object, rwo, target_time);
+
+                if(collision) {
+                    rwo_collison.push_back(rwol.roadway_obstacles[i]);
+                }
+            }
+
+            return rwo_collison;
+        };
+
+        collision_detection::MovingObject CollisionChecking::ConvertRoadwayObstacleToMovingObject(cav_msgs::RoadwayObstacle& rwo){
+
+            collision_detection::MovingObject ro;
+            
+            ro.object_polygon = CollisionChecking::ObjectToBoostPolygon<polygon_t>(rwo.object.pose.pose, rwo.object.size);
+
+            for (auto i : rwo.object.predictions){
+                ro.future_polygons.push_back(CollisionChecking::ObjectToBoostPolygon<polygon_t>(i.predicted_position, rwo.object.size));
+            }
+
+            ro.linear_velocity = rwo.object.velocity.twist.linear;
+
+            return ro;
+        };
+
+        collision_detection::MovingObject CollisionChecking::ConvertVehicleToMovingObject(cav_msgs::TrajectoryPlan& tp, geometry_msgs::Vector3& size, geometry_msgs::Twist& veloctiy){
+            
+            collision_detection::MovingObject v;
+
+            geometry_msgs::Pose pose;
+            pose.position.x = tp.trajectory_points[0].x;
+            pose.position.y = tp.trajectory_points[0].y;
+
+            Eigen::Vector2d vehicle_vector = {tp.trajectory_points[0].x - tp.trajectory_points[1].x , tp.trajectory_points[0].y - tp.trajectory_points[1].y};
+
+            Eigen::Vector2d x_axis = {1, 0};
+
+            double yaw = std::acos(vehicle_vector.dot(x_axis)/(vehicle_vector.norm() * x_axis.norm()));
```

Comment:
vehicle_vector.dot([1, 0]) is always going to give you 
vehicle_vector.x

similarly x_axis.norm() is always going to equal 1.

So this line could be made much simpler and and faster as
```
std::acos(vehicle_vector.x / vehicle_vector.norm())
```

Reply:
It makes scene but  your suggested code and has compile errors.
```
error: invalid operands of types ‘<unresolved overloaded function type>’ and ‘Eigen::GenericNumTraits<double>::Real {aka double}’ to binary ‘operator/’
             std::acos(vehicle_vector.x / vehicle_vector.norm());
```
I think it is better we keep it this way.

Reply:
Try this
```
std::acos(vehicle_vector.x / (double)vehicle_vector.norm())
```
If that doesn't work you can go back to the other method.

Reply:
I added this and again got error. 

error: invalid operands of types ‘<unresolved overloaded function type>’ and ‘Eigen::GenericNumTraits<double>::Real {aka double}’ to binary ‘operator/’
             double yaw = std::acos(vehicle_vector.x / (double)vehicle_vector.norm());